content/dart/concepts/type-conversion/type-conversion.md

```diff
@@ -0,0 +1,496 @@
+  print(stringValue);
+}
+```
+
+The `Shell` output:
+
+```shell
+The value is: 42
+```
+
+The `toString()` method is implicitly called when an object is used in a string interpolation. In Example 2, Dart automatically calls the `toString()` method on `intValue` to obtain the string representation.
+
+---
+
+## Explicit Type Conversion
+
+Explicit type conversion is done by the programmer using predefined functions or methods to convert values from one type to another. The following are the methods for Explicit type conversion:
+
+* `toString()`[^1] : Converts a value to a string.
+* `toDouble()`[^2] : Converts a value to a double.
+* `toInt()`[^3] : Converts a value to an integer.
+* `parse()`[^4] : Converts a string to a numeric type.
+* `tryParse()`[^5]: Similar to `parse()`, but returns `null` if the conversion fails.
+
+---
+
+## toString() Method
+
+In Dart, the `toString()` method plays a significant role in type conversion, especially when converting non-string types to strings. When you explicitly or implicitly convert an object to a string, Dart often relies on the `toString()` method to provide a meaningful string representation of that object.
+
+## toString Method Example 1
+
+```dart
+void main() {
+  var intValue = 42;
+  var stringValue = intValue.toString(); // Explicit toString() call
+
+  print(stringValue);
+
+  var checkCondition = (stringValue.runtimeType == String) ? "$stringValue is a string" : "$stringValue is not a string";
+
+  print(checkCondition);
+}
+```
+
+The `Shell` output:
+
+```shell
+42
+42 is a string
+```
+
+## toString Method Output Variations
+
+For all `double` values of input ranging from 10^-6 (inclusive) to 10^21 (exclusive), the `toString` method returns decimal representation of its input in `string` format. For all other doubles, except for special values like `NaN` or `Infinity`, this method returns an exponential representation using `toStringAsExponential` method.
+
+## Example toString Output 1
+
+```dart
+void main() {
+  var doubleValue1 = 2000000.2/2;
+  var doubleValue2 = 1/2000000;
+
+  var strValue1 = doubleValue1.toString();
+  var strValue2 = doubleValue2.toString();
+
+  print(strValue1);
+  print(strValue2);
+}
+```
+
+The `Shell` output:
+
+```Shell
+1000000.1
+5e-7
+```
+
+For special values, returns `NaN` if the input is `NaN`, returns `Infinity`, if input is `Infinity`, and `-Infinity`, if the input is `-Infinity`.
+
+## Example toString Output 2
+
+```dart
+void main() {
+  var doubleValue1 = 0.0/0.0;
+  var doubleValue2 = 1.0/0.0;
+  var doubleValue3 = -1.0/0.0;
+
+  var strValue1 = doubleValue1.toString();
+  var strValue2 = doubleValue2.toString();
+  var strValue3 = doubleValue3.toString();
+
+  print(strValue1);
+  print(strValue2);
+  print(strValue3);
+}
+```
+
+The `Shell` output:
+
+```Shell
+NaN
+Infinity
+-Infinity
+```
+
+An `int` is converted to a decimal representation with no decimal point.
+
```

Comment:
We don't need any specific function explanation or example in this case. Please remove the above-mentioned lines.